src/Uno.Extensions.Authentication.Oidc/WebAuthenticatorBrowser.cs

```diff
@@ -1,24 +1,30 @@
-﻿
-using IdentityModel.OidcClient.Browser;
+﻿using IdentityModel.OidcClient.Browser;
 using System.Diagnostics;
+using Windows.Foundation;
 
 namespace Uno.Extensions.Authentication.Oidc;
 
 public class WebAuthenticatorBrowser : IBrowser
 {
 	public async Task<BrowserResult> InvokeAsync(BrowserOptions options, CancellationToken cancellationToken = default)
 	{
+		var cts = new CancellationTokenSource(options.Timeout);
+		var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(cts.Token, cancellationToken);
+
 		try
 		{
 #if WINDOWS
-			var userResult = await WinUIEx.WebAuthenticator.AuthenticateAsync(new Uri(options.StartUrl), new Uri(options.EndUrl));
+			var userResult = await WinUIEx.WebAuthenticator.AuthenticateAsync(new Uri(options.StartUrl), new Uri(options.EndUrl), linkedCts.Token);
 			var callbackurl = $"{options.EndUrl}/?{string.Join("&", userResult.Properties.Select(x => $"{x.Key}={x.Value}"))}";
 			return new BrowserResult
 			{
 				Response = callbackurl
 			};
 #else
-			var userResult = await WebAuthenticationBroker.AuthenticateAsync(WebAuthenticationOptions.None, new Uri(options.StartUrl), new Uri(options.EndUrl));
+
+			var userResult = await WebAuthenticationBroker
+				.AuthenticateAsync(WebAuthenticationOptions.None, new Uri(options.StartUrl), new Uri(options.EndUrl))
+				.AsTask(linkedCts.Token);
```

Comment:
I'm still trying to understand how this works, so no big thoughts so far. But when entering the "Authentication: Oidc" test on TestHarness and clicking the Login button I get this exception and the line below to this is not executed on Android and Desktop.
```
fail: Uno.Extensions.Authentication.Oidc.OidcAuthenticationProvider[0]
      Error logging in: System.NotImplementedException: The method or operation is not implemented.
         at Uno.AuthenticationBroker.WebAuthenticationBrokerProvider.AuthenticateAsyncCore(WebAuthenticationOptions options, Uri requestUri, Uri callbackUri, CancellationToken ct) in /__w/1/s/src/Uno.UWP/Security/Authentication/Web/WebAuthenticationBrokerProvider.skia.cs:line 23
         at Uno.AuthenticationBroker.WebAuthenticationBrokerProvider.AuthenticateAsync(WebAuthenticationOptions options, Uri requestUri, Uri callbackUri, CancellationToken ct) in /__w/1/s/src/Uno.UWP/Security/Authentication/Web/WebAuthenticationBrokerProvider.cs:line 65
         at Windows.Foundation.AsyncOperation`1.BuildTaskAsync(FuncAsync`2 taskBuilder) in /__w/1/s/src/Uno.Foundation/Internal/AsyncOperation.TResult.cs:line 70
         at System.WindowsRuntimeSystemExtensions.AsTaskCore[TResult](IAsyncOperation`1 source, CancellationToken ct) in /__w/1/s/src/Uno.UWP/System.Runtime/WindowsRuntimeSystemExtensions.cs:line 66
         at Uno.Extensions.Authentication.Oidc.WebAuthenticatorBrowser.InvokeAsync(BrowserOptions options, CancellationToken cancellationToken) in C:\UnoPlatform\uno.extensions\src\Uno.Extensions.Authentication.Oidc\WebAuthenticatorBrowser.cs:line 25 - (null)
```

On windows I see this:

```
The URI Scheme oidc-auth is not declared in AppxManifest.xml
   at WinUIEx.WebAuthenticator.<Authenticate>d__13.MoveNext() in C:\UnoPlatform\uno.extensions\src\Uno.Extensions.Authentication.UI\WinUIEx_WebAuthenticator.cs:line 229
   at Uno.Extensions.Authentication.Oidc.WebAuthenticatorBrowser.<InvokeAsync>d__0.MoveNext() in C:\UnoPlatform\uno.extensions\src\Uno.Extensions.Authentication.Oidc\WebAuthenticatorBrowser.cs:line 17
```

Would this be something to be looked into?

cc @carldebilly @nickrandolph @kazo0

Reply:
@carldebilly for the windows error mentioned above `The URI Scheme oidc-auth is not declared in AppxManifest.xml ...` I also get it without your changes, from the main branch.